mechanisms/BuildModules.cmake

```diff
@@ -118,7 +118,9 @@ function("make_catalogue")
   endforeach()
   set(${MK_CAT_OUTPUT} ${catalogue_${MK_CAT_NAME}_source} PARENT_SCOPE)
 
-  set_source_files_properties(${catalogue_${MK_CAT_NAME}_source} COMPILE_FLAGS ${ARB_CXXOPT_ARCH})
+  if(${ARB_CXXOPT_ARCH})
+	set_source_files_properties(${catalogue_${MK_CAT_NAME}_source} COMPILE_FLAGS ${ARB_CXXOPT_ARCH})
```

Comment:
the `ARB_CXXOPT_ARCH` variable is used elsewhere in the CMake configureation, so should we explicitly set it to an empty string?
```
bcumming@arapiles:arbor > ag ARB_CXXOPT_ARCH
python/test/cpp/CMakeLists.txt
15:target_compile_options(py_unit PRIVATE ${ARB_CXXOPT_ARCH})

mechanisms/BuildModules.cmake
77:    message("Arbor arch:           ${ARB_CXXOPT_ARCH}")
121:  set_source_files_properties(${catalogue_${MK_CAT_NAME}_source} COMPILE_FLAGS ${ARB_CXXOPT_ARCH})

test/unit/CMakeLists.txt
238:target_compile_options(unit PRIVATE ${ARB_CXXOPT_ARCH})

test/ubench/CMakeLists.txt
29:    target_compile_options(${bench_exe} PRIVATE ${ARB_CXXOPT_ARCH})

test/unit-distributed/CMakeLists.txt
14:target_compile_options(unit-local PRIVATE ${ARB_CXXOPT_ARCH})
22:    target_compile_options(unit-mpi PRIVATE ${ARB_CXXOPT_ARCH})

cmake/arbor-config.cmake.in
52:set(ARB_CXXOPT_ARCH @ARB_CXXOPT_ARCH@)

sup/CMakeLists.txt
10:target_compile_options(arbor-sup PRIVATE ${ARB_CXXOPT_ARCH})

CMakeLists.txt
234:    set_arch_target(ARB_CXXOPT_ARCH "${ARB_ARCH}")
235:    target_compile_options(arbor-private-deps INTERFACE ${ARB_CXXOPT_ARCH})
236:    target_compile_options(arborenv-private-deps INTERFACE ${ARB_CXXOPT_ARCH})
```